src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,232 @@
+    public List<Car> getAllByDriver(Long id) {
+        String query = "SELECT car_id FROM cars_drivers WHERE driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement = connection.prepareStatement(query)) {
+            getAllStatement.setLong(1, id);
+            ResultSet getAllResult = getAllStatement.executeQuery();
+            List<Car> allCars = new ArrayList<>();
+            while (getAllResult.next()) {
+                Long carId = getAllResult.getLong(1);
+                allCars.add(get(carId)
+                        .orElseThrow(() -> new NoSuchElementException("Can`t get car "
+                                + "by id = " + carId)));
+            }
+            return allCars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all car by driver_id " + id, e);
+        }
+    }
+
+    private void insertDrivers(List<Driver> drivers) {
+        String query = "INSERT INTO drivers (name, license_number) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            for (Driver driver : drivers) {
+                statement.setString(1, driver.getName());
+                statement.setString(2, driver.getLicenseNumber());
+                statement.executeUpdate();
+                ResultSet resultSet = statement.getGeneratedKeys();
+                if (resultSet.next()) {
+                    driver.setId(resultSet.getLong(1));
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create drivers "
+                    + drivers, e);
+        }
+    }
+
+    private List<Driver> getDrivers(Long id) {
+        String query = "SELECT d.id, name, license_number FROM cars_drivers cd "
+                + "JOIN drivers d ON cd.driver_id = d.id "
+                + "WHERE d.is_deleted = false AND cd.car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversStatement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            getDriversStatement.setLong(1, id);
+            ResultSet resultDrivers = getDriversStatement.executeQuery();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultDrivers.next()) {
+                drivers.add(createDriver(resultDrivers));
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can`t get drivers by id = " + id, e);
+        }
+    }
+
+    private Car createCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getLong(1));
+        car.setModel(resultSet.getString(2));
```

Comment:
let's use column names instead of indexes
the same for drivers and manufacturers